src/views/List.jsx

```diff
@@ -1,57 +1,81 @@
 import { ListItem } from '../components';
 import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 export function List({ data }) {
-	const [searchInput, setSearchInput] = useState('');
+	const navigate = useNavigate();
 
-	const listItemsToDisplay = data.map((item) => {
-		const isItemInSearch = item.name
-			?.toLowerCase()
-			.includes(searchInput.toLowerCase());
+	const WelcomePrompt = () => {
+		return (
+			<>
+				<h3>Shopping list is currently empty...</h3>
+				<button
+					type="button"
+					onClick={() => {
+						navigate('/add-item');
+					}}
+				>
+					Add Item
+				</button>
+			</>
+		);
+	};
 
-		return isItemInSearch ? <ListItem key={item.id} name={item.name} /> : null;
-	});
+	const FormAndList = () => {
+		const [searchInput, setSearchInput] = useState('');
 
-	const handleKeyDown = (e) => {
-		if (e.keyCode === 13) {
+		const handleKeyDown = (e) => {
+			if (e.keyCode === 13) {
+				e.preventDefault();
+			}
+		};
+		const resetDisplayList = (e) => {
 			e.preventDefault();
-		}
-	};
+			setSearchInput('');
+		};
+
+		const listItemsToDisplay = data.map((item) => {
+			const isItemInSearch = item.name
+				?.toLowerCase()
+				.includes(searchInput.toLowerCase());
 
-	const resetDisplayList = (e) => {
-		e.preventDefault();
-		setSearchInput('');
+			return isItemInSearch ? (
+				<ListItem key={item.id} name={item.name} />
+			) : null;
+		});
```

Comment:
Here is an idea to make this more dry:
```
const listItemsToDisplay = data
    .filter((item) =>
      item.name?
        .toLowerCase()
        .includes(searchInput.toLowerCase())
    )
    .map((item) => <ListItem key={item.id} name={item.name} />);
```

Reply:
Thanks for the suggestion, Yufa. We decided to go as is here.